src/openrct2/audio/Audio.cpp

```diff
@@ -169,12 +169,22 @@ void audio_populate_devices()
     devices.insert(devices.begin(), defaultDevice);
 #endif
 
-    gAudioDeviceCount = static_cast<int32_t>(devices.size());
-    gAudioDevices = Memory::AllocateArray<audio_device>(gAudioDeviceCount);
-    for (int32_t i = 0; i < gAudioDeviceCount; i++)
+    if (devices.size())
```

Comment:
There is a special function to check if the vector is empty. This should already be used throughout our codebase.